transports/quic/src/transport.rs

```diff
@@ -324,8 +331,8 @@ impl Listener {
             match ready!(if_watcher.poll_if_event(cx)) {
                 Ok(IfEvent::Up(inet)) => {
                     let ip = inet.addr();
-                    if self.endpoint.socket_addr.is_ipv4() == ip.is_ipv4() {
-                        let socket_addr = SocketAddr::new(ip, self.endpoint.socket_addr.port());
+                    if self.endpoint.socket_addr().is_ipv4() == ip.is_ipv4() {
```

Comment:
Not sure if it is worth it but we could shorten this by introducing `Endpoint::is_ipv4()`.

Reply:
Is it on purpose that this `if` will yield true if both are IPv6?

`false == false` will yield true and still enter this branch.

Perhaps a more descriptive function could be:

```rust
impl Endpoint {
	fn is_same_ip_family_as(&self, addr: IpAddr) -> bool;
}
```

Reply:
> Is it on purpose that this if will yield true if both are IPv6?

Yes. 

Agreed that it's better to be more descriptive here. I did not add an extra method but as part of e393fe5 I introduced a variable `is_same_ip_family` for this.